tests/integration/common.rs

```diff
@@ -90,6 +90,18 @@ pub fn create_test_store_path() -> std::path::PathBuf {
     temp_file
 }
 
+pub async fn execute_failing_tx(
+    client: &mut TestClient,
+    tx_request: TransactionRequest,
+    expected_error: ClientError,
+) {
+    println!("Executing transaction...");
+    // We compare string since we can't compare the error directly
+    assert_eq!(
+        client.new_transaction(tx_request).err().unwrap().to_string(),
+        expected_error.to_string()
+    );
```

Comment:
Another option is to use the `matches` macro so we don't need to convert both to strings:
```suggestion
        assert_eq!(matches!(client.new_transaction(tx_request), Err(expected_error)))
```

Reply:
I will derive `Eq` to `ClientError`, but thanks!